tests/test_comparison.py

```diff
@@ -0,0 +1,258 @@
+    )
+    assert neq(
+        x=da.from_array(np.array([x])),
+        y=da.from_array(np.array([y])),
+        delta=delta,
+        case_sensitive=case_sensitive,
+    )
+
+
+@pytest.mark.parametrize(
+    "x, min, max, exclude_max, expected",
+    [
+        (1, 0, 1, False, True),
+        (1, 0, 1, True, False),
+        (0.5, 1, 0, False, False),
+        (-0.5, -1, 0, False, True),
+    ],
+)
+def test_between(x, min, max, exclude_max, expected):
+    assert between(x, min, max, exclude_max) == expected
+    assert between(np.array([x]), min, max, exclude_max) == expected
+    assert between(da.from_array(np.array([x])), min, max, exclude_max) == expected
+
+
+@pytest.mark.parametrize("size", [(6, 5, 4, 4)])
+@pytest.mark.parametrize("dtype", [np.float32])
+def test_is(temporal_interval, bounding_box, random_raster_data, process_registry):
+    input_cube = create_fake_rastercube(
+        data=random_raster_data,
+        spatial_extent=bounding_box,
+        temporal_extent=temporal_interval,
+        bands=["B02", "B03", "B04", "B08"],
+        backend="dask",
+    )
+
+    _process = partial(
+        process_registry["is_nan"].implementation,
+        x=ParameterReference(from_parameter="x"),
+    )
+    output_cube = apply(data=input_cube, process=_process)
+    general_output_checks(
+        input_cube=input_cube,
+        output_cube=output_cube,
+        verify_attrs=True,
+        verify_crs=True,
+    )
+    xr.testing.assert_equal(output_cube, xr.zeros_like(input_cube))
+
+    _process = partial(
+        process_registry["is_valid"].implementation,
+        x=ParameterReference(from_parameter="x"),
+    )
+    output_cube = apply(data=input_cube, process=_process)
+    general_output_checks(
+        input_cube=input_cube,
+        output_cube=output_cube,
+        verify_attrs=True,
+        verify_crs=True,
+    )
+    xr.testing.assert_equal(output_cube, xr.ones_like(input_cube))
+
+    _process = partial(
+        process_registry["is_infinite"].implementation,
+        x=ParameterReference(from_parameter="x"),
+    )
+    output_cube = apply(data=input_cube, process=_process)
+    general_output_checks(
+        input_cube=input_cube,
+        output_cube=output_cube,
+        verify_attrs=True,
+        verify_crs=True,
+    )
+    xr.testing.assert_equal(output_cube, xr.zeros_like(input_cube))
+
+
+@pytest.mark.parametrize("size", [(6, 5, 4, 4)])
+@pytest.mark.parametrize("dtype", [np.float32])
+def test_compare(temporal_interval, bounding_box, random_raster_data, process_registry):
+    # TODO: Add test with merge_cubes
+    input_cube = create_fake_rastercube(
+        data=random_raster_data,
+        spatial_extent=bounding_box,
+        temporal_extent=temporal_interval,
+        bands=["B02", "B03", "B04", "B08"],
+        backend="dask",
+    )
+
+    _process = partial(
+        process_registry["eq"].implementation,
+        y=0.5,
```

Comment:
This test has potential to be flaky: If any of the randomly generated data values is 0.5, the test will fail. You could leverage the fact that the random data is always between -100 and 100, so you could make this 200

Reply:
good spot!